libs/util/shared/src/lib/links/link-utils.ts

```diff
@@ -242,6 +250,42 @@ export function getLinkLabel(link: DatasetOnlineResource): string {
   return format ? `${label} (${format})` : label
 }
 
+export async function getLayers(
+  url: string,
+  serviceProtocol: OGCServiceProtocol
+) {
+  const endpointDef = getService(serviceProtocol)
+
+  if (!endpointDef) return undefined
+
+  const endpoint = new endpointDef(url)
+  await endpoint.isReady()
+
+  return endpoint.getLayers()
+}
+
+export function getService(
+  serviceProtocol: OGCServiceProtocol
```

Comment:
can you use the more generic `ServiceProtocol` type here? this would avoid redefining values

Reply:
yes, done